src/planner/match/VertexIdSeek.cpp

```diff
@@ -35,88 +36,67 @@ bool VertexIdSeek::matchNode(NodeContext* nodeCtx) {
         return false;
     }
 
-    auto vidResult = extractVids(matchClauseCtx->where->filter.get());
-    if (!vidResult.ok()) {
+    if (node.alias == nullptr) {
+        // require one named node
         return false;
     }
 
-    nodeCtx->ids = vidResult.value();
-    return true;
-}
-
-StatusOr<const Expression*> VertexIdSeek::extractVids(const Expression* filter) {
-    if (filter->kind() == Expression::Kind::kRelIn) {
-        const auto* inExpr = static_cast<const RelationalExpression*>(filter);
-        if (inExpr->left()->kind() != Expression::Kind::kFunctionCall ||
-            inExpr->right()->kind() != Expression::Kind::kConstant) {
-            return Status::Error("Not supported expression.");
-        }
-        const auto* fCallExpr = static_cast<const FunctionCallExpression*>(inExpr->left());
-        if (*fCallExpr->name() != "id") {
-            return Status::Error("Require id limit.");
-        }
-        auto* constExpr = const_cast<Expression*>(inExpr->right());
-        return constExpr;
-    } else if (filter->kind() == Expression::Kind::kRelEQ) {
-        const auto* eqExpr = static_cast<const RelationalExpression*>(filter);
-        if (eqExpr->left()->kind() != Expression::Kind::kFunctionCall ||
-            eqExpr->right()->kind() != Expression::Kind::kConstant) {
-            return Status::Error("Not supported expression.");
-        }
-        const auto* fCallExpr = static_cast<const FunctionCallExpression*>(eqExpr->left());
-        if (*fCallExpr->name() != "id") {
-            return Status::Error("Require id limit.");
+//    auto vidResult = reverseEvalVids(matchClauseCtx->where->filter.get());
+    VidExtractVisitor vidExtractVisitor;
+    matchClauseCtx->where->filter->accept(&vidExtractVisitor);
+    auto vidResult = vidExtractVisitor.moveVidPattern();
+    if (vidResult.spec != VidExtractVisitor::VidPattern::Special::kInUsed) {
+        return false;
+    }
+    for (auto &nodeVid : vidResult.nodes) {
+        if (nodeVid.second.kind == VidExtractVisitor::VidPattern::Vids::Kind::kIn) {
+            if (nodeVid.first == *node.alias) {
+                nodeCtx->ids = std::move(nodeVid.second.vids);
+                return true;
+            }
         }
-        auto* constExpr = const_cast<Expression*>(eqExpr->right());
-        return constExpr;
-    } else {
-        return Status::Error("Not supported expression.");
     }
+
+    return false;
 }
 
-std::pair<std::string, Expression*> VertexIdSeek::listToAnnoVarVid(QueryContext* qctx,
-                                                                    const List& list) {
+std::pair<std::string, Expression *> VertexIdSeek::listToAnnoVarVid(QueryContext *qctx,
+                                                                    const List &list) {
     auto input = qctx->vctx()->anonVarGen()->getVar();
     DataSet vids({kVid});
     QueryExpressionContext dummy;
```

Comment:
where is dummy used?

Reply:
For constant expression which eval without real context required.